### PR TITLE
bugfix: NPE when there is an application with no routes

### PR DESCRIPTION
**Pull request by dkhacharia**
## Current Situation
If there is a running app which does not have any routes bound to it, we get a NPE.

## Problem Statement
In this case, the whole monitoring is stopped as it is not recoverable and promregator does not do anything.

## Solution Approach
If there are no urls, ignore the application

## Symptom
``` 
java.lang.NullPointerException: Cannot invoke "String.isEmpty()" because the return value of "org.cloudfoundry.promregator.scanner.ReactiveAppInstanceScanner$OSAVector.getDomainId()" is null
	at org.cloudfoundry.promregator.scanner.ReactiveAppInstanceScanner.lambda$10(ReactiveAppInstanceScanner.java:284) ~[classes/:na]
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:386) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxZip$ZipCoordinator.drain(FluxZip.java:756) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxZip$ZipInner.onNext(FluxZip.java:915) ~[reactor-core-3.4.5.jar:3.4.5]
	at reactor.core.publisher.FluxMergeSequential$MergeSequentialMain.drain(FluxMergeSequential.java:432) ~[reactor-core-3.4.5.jar:3.4.5]
```

### CODE REVIEW

File: src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java [hunk 1]
```diff
@@ -251,7 +251,9 @@ public void setNumberOfInstances(int numberOfInstances) {
 				// Set the access url to the selected route (without any protocol or path yet)
 				v.setAccessURL(this.determineApplicationRoute(urls,
 						v.getTarget().getOriginalTarget().getPreferredRouteRegexPatterns()));
-			}
+			} else
+				// if there is no url, skip this one
+				return Mono.empty();
```
Comment (eaglerainbow) [inline src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java:256]:
```suggestion
			} else {
				// if there is no url, skip this one
				return Mono.empty();
			}
```

**Review by eaglerainbow — COMMENTED** [review]
Nice catch.
I just wonder that we don't have a unit test for this (looking into it)

**Comment by eaglerainbow** [comment]
most likely caused by #195

**Comment by eaglerainbow** [comment]
Requires "downport" to 0.9.x plus release of new patch.

**Comment by eaglerainbow** [comment]
Don't forget to merge https://github.com/promregator/promregator/tree/pr203-unittest as well (provides unit tests for this case here)

**Comment by eaglerainbow** [comment]
Hi @dkhacharia 

Thanks for submitting! If you accept the small adjustment of the review, I'd merge this PR (plus some a unit test) for it and downport things to 0.9.x.

**Comment by dkhacharia** [comment]
Thanks for quick response.
Suggestion accepted. Please merge it :)

**Comment by eaglerainbow** [comment]
Thanks for your PR!

**Comment by eaglerainbow** [comment]
89b9af085e1c11610bed1fee509341ff9b34b871 merged into `master`
and double maintained via cherry-pick into `rel-0.9`.

**Pull request merged**